src/cryptoadvance/specter/templates/includes/recipient-box.html

```diff
@@ -0,0 +1,486 @@
+			});			
+
+
+		}
+
+
+
+		markAdddressGreenIfMine(){
+			// mark own addresses green
+			this.isAddressMyOwn().then((isMine) => {
+				this.markRecipient(isMine);
+			});
+		}
+
+		dispatchAddressOnInput(result) {
+			this.markAdddressGreenIfMine();
+			this.dispatchEvent(new CustomEvent('address-input'));
+		}
+
+		dispatchRemove(result) {
+			this.dispatchEvent(new CustomEvent('remove'));
+		}
+
+		dispatchSendMax(result) {
+			this.dispatchEvent(new CustomEvent('send-max'));
+		}
+
+
+
+		// this function together with attributeChangedCallback ensures that setting attributes directly in html 
+		// like <recipient-box title='title'></div>  works.
+		static get observedAttributes() {
+			// define the attributes that can be defined via <recipient-box name="recipient-box" id='4'></recipient-box>	
+			var attributes = ["value", "name", "id", "title", "address", "label", "unit", 
+								"recipientId", "max", "step", "amount", "hiddenRemoveButton"];
+			return attributes;
+		}
+
+		attributeChangedCallback(attribute, oldValue, newValue) {
+			if (oldValue == newValue){return}
+			if (attribute=="value"){
+				this.value = newValue;
+			} else if (attribute=="name"){
+				this.name = newValue;
+			} else if (attribute=="address"){
+				this.address = newValue;
+			} else if (attribute=="title"){
+				this.title = newValue;
+			} else if (attribute=="recipientId"){
+				this.recipientId = newValue;
+			} else if (attribute=="label"){
+				this.label = newValue;
+			} else if (attribute=="unit"){
+				this.unit = newValue;
+			} else if (attribute=="step"){
+				this.step = newValue;
+			} else if (attribute=="max"){
+				this.max = newValue;
+			} else if (attribute=="amount"){
+				this.amount = newValue;
+			} else if (attribute=="id"){
+				this.id = newValue;
+			} else if (attribute=="hiddenRemoveButton"){
+				this.hiddenRemoveButton = newValue;
+			}
+			
+		}
+
+
+
+		// title
+		set title(newValue){
+			this.titleElement.innerText = newValue;
+		}			
+		get title(){
+			return this.titleElement.innerText;
+		}		
+
+		// address
+		set address(newValue){
+			this.addressElement.value = newValue;
+		}			
+		get address(){
+			return this.addressElement.value;
+		}		
+
+		// label
+		set label(newValue){
+			this.labelElement.value = newValue;
+		}			
+		get label(){
+			return this.labelElement.value;
+		}		
+
+		// unit
+		set unit(newValue){
+			// this function is different for bitcoin and liquid 
+			this.setUnit(newValue);
+		}			
+		get unit(){
+			// this function is different for bitcoin and liquid 
+			return this.getUnit();
+		}				
+
+		// step
+		set step(newValue){
+			this.amountElement.step = newValue;
+		}			
+		get step(){
+			return parseFloat(this.amountElement.step );
+		}		
+
+		// amount
+		set amount(newValue){
+			this.amountElement.value = newValue;
+			this.calculateConvertedUnit()
+		}			
+		get amount(){
+			var value = parseFloat(this.amountElement.value );
+			if (!value) {value = 0}
```

Comment:
If you always set the value to 0 here, this will never lead to this part in `calculateConvertedUnit()`
```
			if (convertedAmount == 'NaN') {
				convertedAmount = '-'
			}
```
This also leads to confusing error messaging when submitting:

<img width="510" alt="grafik" src="https://user-images.githubusercontent.com/70536101/181487648-7442b9a1-914a-4226-88ac-ddfdd2e65b85.png">

We have a pretty encompassing `validateAmount()` in `wallet_send`that is taking care of all weird input. I would just delete` if (!value) {value = 0}`.

Reply:
Ok.